lib/kubeclient/common.rb

```diff
@@ -87,11 +87,18 @@ def initialize_client(
       @http_max_redirects = http_max_redirects
       @as = as
 
-      if auth_options[:bearer_token]
-        bearer_token(@auth_options[:bearer_token])
-      elsif auth_options[:bearer_token_file]
+      @log = Logger.new(STDOUT)
+      @log.formatter = proc do |severity, datetime, progname, msg|
+          "#{datetime} [#{severity}]: #{msg}\n"
+      end
+
+      if auth_options[:bearer_token_file]
         validate_bearer_token_file
+        @log.info("Reading bearer token from #{@auth_options[:bearer_token_file]}")
         bearer_token(File.read(@auth_options[:bearer_token_file]))
+      elsif auth_options[:bearer_token]
+        bearer_token(@auth_options[:bearer_token])
+        @log.info("bearer_token_file path not provided. Kubeclient will not be able to refresh the token if it expires")
```

Comment:
Is there any strong reason to add this log line? We are using kubeclient in a CLI app where the bearer token is refreshed through a process outside kubeclient. Having this line messes up our CLI logging now. Does it make sense to make it atleast a debug level?

Reply:
Good point.  
It seems this is first place where kubeclient is logging at all on v4.y branch.
The similar-but-different implementation on master branch won't have these logs (though it has some other logging in new Informer class, including error level).

=> I tend to agree by default these should not be visible.  
=> Downgrading both to .debug level sounds fine to me.  
=> Other options to control logging, e.g. injecting a Logger object, are also possible but I think we can defer such over-thinking to master branch and future 5.y releases...

@harshit-splunk do you have an opinion on this?  
@rvignesh89 Could you send a PR?

Reply:
Hmm I see debug is still visible by default:
```
$ ruby -e 'require "logger"; Logger.new(STDOUT).debug("foo")'
D, [2022-09-04T15:15:12.509754 #747358] DEBUG -- : foo
```
I guess it's better for back-compat (at least on 4.y versions) that logging be opt-in, and by default kubecliet remain silent.  WDYT?

In this case the logging is merely giving (useful) advice on parameters passed to constructor.  _Technically_, if we just dropped them, the caller could easily implement similar functionality.  

Though once they become aware of the idea these messages suggest, they don't even need them logged.  It's similar to a deprecation message (which suggests it should be at most printed once), except that we're not really deprecating `:bearer_token`, it remains useful.

Reply:
Hmm, I see we forgot to mention the new refresh functionality in README.md.  
=> I'm thinking we could drop these messages altogether and replace them with explanation of the benefit in README — but would like to hear @harshit-splunk's opinion.

Reply:
I think we should remove the logs since other components don't log. I only added those to debug this functionality. I should have removed after testing.

Reply:
I'd vote for removing logging or at least have the opportunity to inject/config the logger. In my case'd like to silence the log as it 'pollutes' STDOUT which I need for other reasons

Reply:
btw. my (hopefully temporary) hack is to monkey patch for a local`STDOUT` logging into the void....

```
module Kubeclient
  module ClientMixin
    STDOUT = StringIO.new
  end
end
```

Reply:
Is there open work to remove this line? I agree it should either not be in stdout or be classified as debug logging.

Reply:
I have created a PR(https://github.com/ManageIQ/kubeclient/pull/577)

Reply:
^ Thanks. Merged and released 4.10.1.  Also added some words in the README about `bearer_token` vs `bearer_token_file` (#579).

Reply:
@cben I see 4.10.1 in Rubygems but not reflected here, is there a reason why? https://github.com/ManageIQ/kubeclient/blob/master/lib/kubeclient/version.rb

Reply:
Nevermind I see there is a long running branch for the 4.x release here https://github.com/ManageIQ/kubeclient/blob/v4.y/CHANGELOG.md

Might be worth adding a note to the readme so folks know where to find the latest release notes